ui/src/app/applications/components/application-details/application-details.tsx

```diff
@@ -512,10 +512,13 @@ export class ApplicationDetails extends React.Component<RouteComponentProps<{nam
 
     private async updateApp(app: appModels.Application) {
         const latestApp = await services.applications.get(app.metadata.name);
+        const latestAppSyncPolicy = latestApp.spec?.syncPolicy;
         latestApp.metadata.labels = app.metadata.labels;
         latestApp.metadata.annotations = app.metadata.annotations;
         latestApp.spec = app.spec;
-        const updatedApp = await services.applications.update(latestApp);
+        const updatedApp = await services.applications.update(latestApp, {
+            validate: JSON.stringify(latestAppSyncPolicy) === JSON.stringify(app.spec?.syncPolicy)
```

Comment:
As I understand that we are relying on the fact that we never change the sync policy together with other fields. So if sync policy has changed then we can assume it is the only updated field and we can skip validation, right ?

User might change both sync policy and other spec fields using the manifest tab (e.g. https://cd.apps.argoproj.io/applications/argo-cd?resource=&node=argoproj.io%2FApplication%2Fargocd%2Fargo-cd%2F0&tab=manifest) so we might accidently skip validation.


I would suggest adding an optional `validate` parameter to the `updateApp` method with `true`  as the default value. The `setAutoSync` and `unsetAutoSync` can explicitly set `validate` to false.